theme/scripts/productDescription.ts

```diff
@@ -0,0 +1,83 @@
+export default(function() {
+	const ProductDescription = {
+		SELECTORS: {
+			tab: "[data-product-description-tab]",
+      content: "[data-product-description-content]",
+      tabsWrapper: ".ProductDescription__tabs",
+      nav: ".Nav"
+		},
+    CLASSES: {
+      activeTab: "ProductDescription__button--active"
+    },
+
+		init() {
+			const tabs = [].slice.call(
+        document.querySelectorAll(ProductDescription.SELECTORS.tab)
+      );
+
+      const tabbedContent = [].slice.call(
+        document.querySelectorAll(ProductDescription.SELECTORS.content)
+      );
+
+      const tabsWrapper: HTMLElement = document.querySelector(ProductDescription.SELECTORS.tabsWrapper);
+
+      const navs = [].slice.call(
+        document.querySelectorAll(ProductDescription.SELECTORS.nav)
+      );
+
+			if (!tabs || !tabbedContent || !tabsWrapper || !navs) return;
+
+      let height = ProductDescription.getStickyHeight(tabsWrapper, navs);
+      window.addEventListener('resize', () => {
+        height = ProductDescription.getStickyHeight(tabsWrapper, navs);
+      })
+      
+      window.addEventListener('scroll', () => {
+        window.requestAnimationFrame(() => {
+          ProductDescription.scrollHandler(tabs, tabbedContent, height);
+        });
+      })
+		},
+
+    getStickyHeight(tabsWrapper: HTMLElement, navs: HTMLElement[]): number {
+      const tabsWrapperHeight = tabsWrapper.offsetHeight;
+      const activeNav = navs.find((elem) => {
+        if (elem.offsetHeight !== 0) return elem;
+      })
+      return tabsWrapperHeight + activeNav.offsetHeight + 16;
```

Comment:
What's `16` in this case? We generally like to store these type of loose integers in a named constant so that it's a bit clearer what it represents. You can declare this outside the module, and we like to use all caps e.g. `const STICKY_HEIGHT_BUFFER = 16;`

Reply:
the 16 is just some sort of buffer that ended up having the best experience in terms of when the active state switches from one tab to the other. I chose 16 just because it's 1 rem. It's a bit finicky because I'm sure any number between 10-20px would work fine. 

I'll make it a constant and add a comment.